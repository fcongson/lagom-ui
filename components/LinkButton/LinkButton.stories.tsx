import { LinkButton } from "./LinkButton";

export default {
  title: "LinkButton",
  component: LinkButton,
};

export const Link = () => <LinkButton href="#">Link Button</LinkButton>;
