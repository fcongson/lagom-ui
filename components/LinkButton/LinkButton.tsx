import styled from "styled-components";
import { Button } from "../Button";

const StyledLink = styled.a`
  text-decoration: none;
`;

interface LinkButtonProps extends React.ComponentPropsWithoutRef<"button"> {
  /**
   * Choose between the primary and secondary variants
   */
  primary?: boolean;
  /**
   * Where to navigate to
   */
  to?: string;
}

export const LinkButton: React.FunctionComponent<LinkButtonProps> = ({
  children,
  primary = false,
  to,
  ...restProps
}) => (
  <StyledLink href={to}>
    <Button primary={primary} {...restProps}>
      {children}
    </Button>
  </StyledLink>
);
