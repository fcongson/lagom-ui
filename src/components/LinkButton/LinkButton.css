/* @import "../Button/Button.css"; */

.lagom-link-button {
  display: contents;
}
