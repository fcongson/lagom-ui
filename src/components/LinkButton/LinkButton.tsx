import clsx from "clsx";
import styled, { css } from "styled-components";
import { Button } from "../Button";

const linkCss = css`
  &.lagom-link-button {
    display: contents;
  }
`;

const StyledLink = styled.a`
  ${linkCss}
`;

interface LinkButtonProps extends React.ComponentPropsWithoutRef<"button"> {
  /**
   * Classname(s) applied to this component
   */
  className?: string;
  /**
   * Choose between the primary and secondary variants
   */
  primary?: boolean;
  /**
   * Where to navigate to
   */
  to: string;
  /**
   * A framework specific link component
   */
  Component?: React.ComponentType<{ to: string; children?: React.ReactNode }>;
}

export const LinkButton: React.FunctionComponent<LinkButtonProps> = ({
  children,
  className,
  primary = false,
  to,
  Component,
  ...restProps
}) => {
  if (Component) {
    const StyledComponent = styled(Component)<{ className?: string }>`
      ${linkCss}
    `;
    return (
      <StyledComponent className={clsx(`lagom-link-button`, className)} to={to}>
        <Button primary={primary} {...restProps}>
          {children}
        </Button>
      </StyledComponent>
    );
  }

  return (
    <StyledLink className={clsx(`lagom-link-button`, className)} href={to}>
      <Button primary={primary} {...restProps}>
        {children}
      </Button>
    </StyledLink>
  );
};
