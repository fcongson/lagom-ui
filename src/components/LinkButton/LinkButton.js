import { clsx } from "clsx";
import React from "react";
import { Button } from "../Button/Button.js";

import "./LinkButton.css";

export const LinkButton = ({
  children,
  className,
  to,
  Component,
  ...restProps
}) =>
  !!Component ? (
    <Component className={clsx(`lagom-link-button`, className)} to={to}>
      <Button {...restProps}>{children}</Button>
    </Component>
  ) : (
    <a className={clsx(`lagom-link-button`, className)} href={to}>
      <Button {...restProps}>{children}</Button>
    </a>
  );
