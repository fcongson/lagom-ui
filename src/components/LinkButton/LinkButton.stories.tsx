import { LinkButton } from "./LinkButton";

export default {
  title: "Components/Link Button",
  component: LinkButton,
};

export const Primary = () => (
  <LinkButton to="/?path=/story/components-link-button--primary" primary>
    Link button primary
  </LinkButton>
);

export const Secondary = () => (
  <LinkButton to="/?path=/story/components-link-button--secondary">
    Link button secondary
  </LinkButton>
);
