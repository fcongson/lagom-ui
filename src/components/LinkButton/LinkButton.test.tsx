import { render, screen } from "@testing-library/react";
import { LinkButton } from "./LinkButton";

describe("LinkButton", () => {
  test("Renders", () => {
    render(<LinkButton to="#">Text</LinkButton>);
    expect(screen.getAllByRole("button")).toMatchSnapshot();
  });
  test("Renders primary", () => {
    render(
      <LinkButton to="#" primary>
        Text
      </LinkButton>
    );
    expect(screen.getAllByRole("button")).toMatchSnapshot();
  });
  test("Renders secondary", () => {
    render(
      <LinkButton to="#" primary={false}>
        Text
      </LinkButton>
    );
    expect(screen.getAllByRole("button")).toMatchSnapshot();
  });
});
