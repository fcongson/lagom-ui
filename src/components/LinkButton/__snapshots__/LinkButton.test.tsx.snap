// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton Renders 1`] = `
[
  <button
    class="sc-bdfCDU gPECCf lagom-button lagom-button--secondary"
  >
    Text
  </button>,
]
`;

exports[`LinkButton Renders primary 1`] = `
[
  <button
    class="sc-bdfCDU gPECCf lagom-button lagom-button--primary"
  >
    Text
  </button>,
]
`;

exports[`LinkButton Renders secondary 1`] = `
[
  <button
    class="sc-bdfCDU gPECCf lagom-button lagom-button--secondary"
  >
    Text
  </button>,
]
`;
